nix/searxng.nix

```diff
@@ -0,0 +1,94 @@
+{ pkgs
+, lib
+, name
+, config
+, ...
+}:
+let
+  inherit (lib) types;
+  yamlFormat = pkgs.formats.yaml { };
+
+  settingType =
+    with types;
+    (oneOf [
+      bool
+      int
+      float
+      str
+      (listOf settingType)
+      (attrsOf settingType)
+    ])
+    // {
+      description = "JSON value";
+    };
+in
+{
+  options = {
+    enable = lib.mkEnableOption name;
+    package = lib.mkPackageOption pkgs "searxng" { };
+
+    host = lib.mkOption {
+      description = "Searxng bind address";
+      default = "127.0.0.1";
+      type = types.str;
+    };
+
+    port = lib.mkOption {
+      description = "Searxng port to listen on";
+      default = 8080;
+      type = types.port;
+    };
+
+    settings = lib.mkOption {
+      types = yamlFormat.type;
+      default = {
+        server.secret_key = "secret";
+        server.limiter = false;
+      };
+      example = lib.literalExpression ''
+        {
+          server.secret_key = "secret";
+          server.limiter = false;
+        }
+      '';
+      description = ''
+        Searxng settings. These will be merged with (taking precedence over)
+        the default configuration.
```

Comment:
I don't understand the merge semantics here. What is the "default configuration"? If it is referring to lines 44-47, I don't see where the merge happens.

Reply:
I removed the obsolete description